docs/site/tutorials/authentication/Quick-Authentication-Tutorial-Session-Tokens.md

```diff
@@ -0,0 +1,297 @@
+        `User with clientKey ${credentials.clientKey} not found.`,
+      );
+    }
+
+    // Using the password hasher from https://bit.ly/34q3IJp
+    const passwordMatched = await this.passwordHasher.comparePassword(
+      credentials.clientSecret,
+      foundUser.password,
+    );
+
+    if (passwordMatched) {
+      // We want to return the user to the controllers
+      // But we must include a securityId
+      foundUser[securityId] = foundUser.id;
+      return foundUser;
+    } else {
+      throw new HttpErrors.Unauthorized('The credentials are not correct.');
+    }
+    */
+  }
+
+  extractCredentials(request: Request): Credentials {
+    if (!request.headers.authorization) {
+      throw new HttpErrors.Unauthorized(`Authorization header not found.`);
+    }
+
+    // For example: 'Basic ' + base64encoded('username:password')
+    //          or: 'Basic ' + base64encoded('email@address.com:secret')
+    const authHeaderValue = request.headers.authorization;
+
+    const headerParts = authHeaderValue.split(' ');
+    if (headerParts[0] !== 'Basic') {
+      throw new HttpErrors.Unauthorized(
+        `Authorization header is not of type 'Basic'.`,
+      );
+    }
+    if (headerParts.length !== 2) {
+      throw new HttpErrors.Unauthorized(
+        `Authorization header should follow 'Basic' with a single base64 encoded string`,
+      );
+    }
+
+    const encodedAuth = headerParts[1];
+    const decodedAuth = new Buffer(encodedAuth, 'base64').toString('utf8');
+
+    const authParts = decodedAuth.split(':');
+    if (authParts.length !== 2) {
+      throw new HttpErrors.Unauthorized(
+        `Authorization header's decoded value should have exactly two parts, separated by a ':'`,
+      );
+    }
+
+    return {clientKey: authParts[0], clientSecret: authParts[1]};
+  }
+}
+```
+
+## Register the Basic authentication strategy
+
+Now we need to tell our application that this strategy exists and can be used.
+
+Open `src/application.ts` and add the following lines:
+
+```ts
+import {
+  AuthenticationComponent,
+  registerAuthenticationStrategy,
+} from '@loopback/authentication';
+import {BasicAuthenticationStrategy} from './strategies/basic-strategy';
+
+// At the bottom of the constructor function
+
+this.component(AuthenticationComponent);
+registerAuthenticationStrategy(this, BasicAuthenticationStrategy);
+```
+
+## Start using the Basic authentication strategy in controllers
+
+To employ our authentication in controllers, we can use the `@authenticate()`
+decorator.
+
+For OAuth2 you may want to create a `/users/login` route. You could add it to
+your existing UserController (or to the PingController, if that is all you have
+right now).
+
+Add the following lines to your controller:
+
+```ts
+import { authenticate, AuthenticationBindings } from '@loopback/authentication';
+
+// In the constructor's arguments
+
+    @inject(AuthenticationBindings.CURRENT_USER, { optional: true }) private user: UserProfile & User,
```

Comment:
The reason I put `UserProfile & User` in the tutorial is that we found it pretty helpful in our applications to pass the entire User model to the controllers. So I wanted to show readers that is an option available to them.

But the demo cannot do this because there is no User model or repository! That's why I just left it out of my own PingController.

My hope was that people doing the tutorial will delete the `User` if they don't have a collection, or import it if they do have a collection.

Resolution: I could drop back to just `UserProfile` and add a comment:
```ts
// Put UserProfile & User there, if you want to pass the full user model to your controllers
```
Does that sound ok?

Reply:
Let's remove this comment altogether.

We are using this trick to pass the entire User model to controllers, but that's out of the scope of this tutorial.